api/src/domain/types.ts

```diff
@@ -61,6 +68,45 @@ export type EncryptTaxId = (userData: UserData) => Promise<UserData>;
 export type AddNewsletter = (userData: UserData) => Promise<UserData>;
 export type AddToUserTesting = (userData: UserData) => Promise<UserData>;
 
+export interface AccessTokenClient {
+  getAccessToken: () => Promise<string>;
```

Comment:
[dust/sand] This is an idea for a follow-up refactor for sure, but hear me out:

I think almost all of these type definitions (for AccessTokenClient, BusinessAddressClient, etc) do not belong here in domain. I think they should live in a `types` file within the `client` folder specifically. Because they should be scoped to that folder and that folder alone. The app at large does NOT have a concept of an AccessTokenClient, and I don't think it should. It's a sub-concept of how the DynamicsLicenseStatusClient works - it's not its own thing in isolation.

From there, you could actually _cement_ that idea in the architecture of the code. You could do this by putting these types and all the "sub-clients" for Dynamics in their own subfolder inside of `/client`. And then you can use the DependencyChecker to explicitly prevent any imports from anywhere else in the code other than the `client` folder and the `app.ts` setup.

What this would do is concretely define that these are NOT types that are available to the application interface. By having them in the domain here, we're basically saying that any domain code could in theory make a call to a `getAccessToken` function on a client if it's provided a client of that type. That isn't really accurate because these should not be functions that are "available" to anything in the domain - they should only be available to the stuff in Client, because that's their purpose of existing.

I might be off-base on this idea, but I'm happy to talk more about it if you want.

Reply:
@FarazA22 and I talked about this and we want to bring it up to the team at large because there are tradeoffs to the different solutions we came up with and there is value to getting everyones input.